packages/eslint-plugin/src/utils/property-selector.ts

```diff
@@ -96,6 +96,21 @@ export const reportTypeError = (
   });
 };
 
+export const reportError = (
```

Comment:
Hmmmm, I don't know if this `function` should reside here as it's specific for `@Component`. I'd suggest to move the `context.report(...)` directly to the `if` statement in (without creating a `function`) `component-selector.ts` file (see the comment there).